app/src/operations_endpoints.h

```diff
@@ -0,0 +1,402 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "app_data.h"
+#include "historical/historical_queries_adapter.h"
+#include "visit_each_entry_in_value.h"
+
+#include <ccf/json_handler.h>
+#include <ccf/node_context.h>
+#include <ccf/odata_error.h>
+#include <ccf/service/tables/nodes.h>
+
+namespace scitt
+{
+  using OperationCallback = std::function<void(
+    ccf::endpoints::EndpointContext&, const nlohmann::json&, nlohmann::json&&)>;
+
+  /**
+   * An indexing strategy which maintains a map from Operation ID to state.
+   *
+   * The state can be one of "pending", "failed" or "registered". In the last
+   * case, the transaction ID which completed the operation is recorded.
+   *
+   * TODO: this should flush old entries to avoid infinite memory growth. Either
+   * we give it a fixed number of entries (a few Ks), or a fixed duration (about
+   * an hour). There's no reason why we should keep these around for longer:
+   * clients are supposed to exchange the operation ID for the entry ID of the
+   * result shortly after the operation completes.
+   */
+  class OperationsIndexingStrategy
+    : public VisitEachEntryInValueTyped<OperationsTable>
+  {
+  public:
+    OperationsIndexingStrategy() : VisitEachEntryInValueTyped(OPERATIONS_TABLE)
+    {}
+
+    struct OperationState
+    {
+      OperationStatus status;
+      std::optional<ccf::TxID> completion_tx;
+      std::optional<nlohmann::json> error;
+    };
+
+    std::optional<OperationState> lookup(const ccf::TxID& operation_id) const
+    {
+      std::lock_guard guard(lock);
+      auto it = operations.find(operation_id.seqno);
```

Comment:
`ccf::TxID` doesn't have a defined comparison operator (probably for good reasons), which means it can't be used in an `std::map`.

Keys in the `std::map` always refer to committed transactions, so their sequence number is guaranteed to be unique. I guess there should be a check on lookup though to make sure the view is correct.